netket/sampler/metropolis_hastings.py

```diff
@@ -110,7 +110,7 @@ def machine(self, machine):
     def reset(self, init_random=False):
         if init_random:
             for state in self._state:
-                self._hilbert.random_vals(state, random_engine())
+                self._hilbert.random_vals(out=state)
```

Comment:
This means that Python samplers will work only with Python Hilbert spaces, right?

Reply:
Not really, also c++ hilbert spaces have a method random_vals with default values for random_engine, I think it should work also in that case

Reply:
Is this covered by one of our unit tests? If not, it would be good to add that test I think.